doc/04-Upgrading.md

```diff
@@ -3,6 +3,49 @@
 Specific version upgrades are described below. Please note that version upgrades are incremental.
 If you are upgrading across multiple versions, make sure to follow the steps for each of them.
 
+## Upgrading to Icinga DB v1.1.2
+
+Please apply the `1.1.2.sql` upgrade script to your database.
+For package installations, you can find this file at `/usr/share/icingadb/schema/mysql/upgrades/` or
+`/usr/share/icingadb/schema/pgsql/upgrades/`, depending on your database type.
+
+As the daemon checks the schema version, the recommended way to perform the upgrade is to stop the daemon, apply the
+schema upgrade and then start the new daemon version. If you want to minimize downtime as much as possible, it is safe
+to apply this schema upgrade while the Icinga DB v1.1.1 daemon is still running and then restart the daemon with the
+new version. Please keep in mind that depending on the distribution, your package manager may automatically attempt to
+restart the daemon when upgrading the package.
+
+### Upgrading the state_history Table (optional)
+
+Icinga DB 1.1.1 already handles hosts/services' `max_check_attempts` above 255.
+But actual soft state changes with check attempt above 255 in your history will crash Icinga DB,
+even after upgrading to v1.1.2. The `check_attempt` column of the `state_history` table is too small
+to fit larger values. ([#655](https://github.com/Icinga/icingadb/issues/655))
```

Comment:
> Icinga DB 1.1.1 already handles hosts/services' `max_check_attempts` above 255.

I'm not even sure what this is trying to tell me here. Why 1.1.1 in particular? Also, I don't think it's relevant here whether some other version would have worked with `max_check_attempts > 255` as long as nothing ever reached that number of check attempts.

> But actual soft state changes with check attempt above 255 in your history will crash Icinga DB, even after upgrading to v1.1.2.

That sounds rather worrisome to me without any extra context given at that point.

Maybe simply start by explaining that previous versions used an insufficient datatype to store the check attempt in the history and that this is fixed for new installations that import the full schema, but it's not automatically fixed by the regular schema upgrade because [insert existing explanation here].

Reply:
1. v1.1.1 is obviously the previous version. Also, I can't just say that large max_check_attempts didn't work, because they did. This paragraph precisely describes the issue for informed decision purposes.
2. If we prefer the user to apply the upgrade, we should reason it. A crash is a pretty good reason.

Reply:
> Also, I can't just say that large max_check_attempts didn't work, because they did.

Well, technically it's not just the `max_check_attempts` value that causes the problem, you can configure a high value there without immediately/certainly running in a problem. But I wouldn't say high `max_check_attempts` are working if Icinga DB crashes if the maximum is actually reached. Also, you could use wording like "checkable reaches a check attempt higher than 255". That would accurately describe it. I don't think "but technically you could configure `max_check_attempts` over 9000 on an always OK check and it would work fine" is necessary information nor does it help. Why would you increase `max_check_attempts` if you don't expect it to reach that value unless you're testing for bugs?

> If we prefer the user to apply the upgrade, we should reason it. A crash is a pretty good reason.

I'm not saying that you shouldn't mention it at all. It's just that if you read that section top to bottom, with the information you were given up to that sentence, you may ask yourself "well, if it still crashes, what does this update even do?".